Apps/AccountDataAccess/src/Patient/PatientRepository.cs

```diff
@@ -91,193 +76,104 @@ public async Task<PatientQueryResult> Query(PatientQuery query, CancellationToke
-                                    await this.GetDemographicsAsync(OidType.Phn, phn).ConfigureAwait(true);
-
-            return new PatientQueryResult(new[] { patient! });
+            return await this.blockedAccessDelegate.GetDataSourcesAsync(hdid).ConfigureAwait(true);
         }
 
-        private async Task<PatientModel?> GetPatientIdentityAsync(string hdid)
+        private static string GetStrategyKey(string? hdid, PatientDetailSource source)
         {
-            try
-            {
-                PatientIdentity result = await this.patientIdentityApi.GetPatientIdentityAsync(hdid).ConfigureAwait(true);
-                PatientModel patient = this.mapper.Map<PatientModel>(result);
-                this.CachePatient(patient);
-                return patient;
-            }
-            catch (ApiException e) when (e.StatusCode == HttpStatusCode.NotFound)
-            {
-                this.logger.LogInformation("PHSA could not find patient identity for {Hdid}", hdid);
-                return null;
-            }
+            string prefix = "HealthGateway.AccountDataAccess.Patient.Strategy.";
+            string type = hdid != null ? "Hdid" : "Phn";
+            string suffix = "Strategy";
+            return $"{prefix}{type}{source}{suffix}";
         }
 
-        /// <summary>
-        /// Attempts to get the Patient model from the Generic Cache.
-        /// </summary>
-        /// <param name="identifier">The resource identifier used to determine the key to use.</param>
-        /// <param name="identifierType">The type of patient identifier we are searching for.</param>
-        /// <returns>The found Patient model or null.</returns>
-        private PatientModel? GetFromCache(string identifier, PatientIdentifierType identifierType)
+        private async Task<PatientQueryResult> Handle(PatientDetailsQuery query, CancellationToken ct)
         {
-            using Activity? activity = Source.StartActivity();
-            PatientModel? retPatient = null;
-            if (this.cacheTtl > 0)
-            {
-                switch (identifierType)
-                {
-                    case PatientIdentifierType.Hdid:
-                        this.logger.LogDebug("Querying Patient Cache by HDID");
-                        retPatient = this.cacheProvider.GetItem<PatientModel>($"{PatientCacheDomain}:HDID:{identifier}");
-                        break;
+            this.logger.LogDebug("Patient details query source: {Source} - cache: {Cache}", query.Source, query.UseCache);
 
-                    case PatientIdentifierType.Phn:
-                        this.logger.LogDebug("Querying Patient Cache by PHN");
-                        retPatient = this.cacheProvider.GetItem<PatientModel>($"{PatientCacheDomain}:PHN:{identifier}");
-                        break;
-                }
+            if (ct.IsCancellationRequested)
```

Comment:
Should be removed. The token should propogate all the way down to the db context, or let it be for now

Reply:
This would require some argument drilling, many of the subsequent methods called to implement this section of the strategies don't accept a cancellation token. so I would need to update many of the interfaces and method calls.

Should we not get an MVP out and re-approach the optimization in a new ticket. there is quite a lot of the tasks downstream still blocked as we don't have a working backend. this could be optimized in parallel with the front end development.

However if this is a non-starter, please let me know I'll start tackling it

Reply:
I don't think this requires anything but removing the `if`

Reply:
I wish it was as simple as that,

This is the last point in the code before it get passed to the strategies, which consumes either a refit API or client registeries delegate.

removing the if statement causes not used variable errors, which spirals into passing the token through all the strategies.